fuzz_lightyear/supplements/factory.py

```diff
@@ -70,16 +88,36 @@ def register_factory(keys: Union[str, Iterable[str]]) -> Callable:
     """
     # This is renamed just to make mypy happy.
     _keys = listify_decorator_args(keys)
+    _endpoints = listify_decorator_args(endpoints)
 
     def decorator(func: Callable) -> Callable:
         wrapped = inject_user_defined_variables(func)
 
         mapping = get_user_defined_mapping()
         for key in _keys:
             if key in mapping:
```

Comment:
nit: flip this logic, so that we handle the case when `key not in mapping` first, because that's what is handled during the first calls to `register_factory`. The latter case is only executed with multiple `register_factory` decorators are used, with the same key.